src/ir/auth_logic/ast_equality_test.cc

```diff
@@ -61,4 +61,206 @@ TEST(AstEqualityTestSuite, Test7) {
   EXPECT_TRUE(!(p1 == p2));
 }
 
+TEST(AstEqualityTestSuite, PrinTest) {
```

Comment:
There is a better idiom for writing equality tests that is quite exhaustive.  Can we change to that pattern?

e.g., 

https://github.com/google-research/raksha/blob/258b6d96db68a8ab79e04d6817270bf5a3167f8d/src/ir/value_test.cc#L162

Reply:
fixed